src/SearchParameters/index.js

```diff
@@ -704,6 +704,43 @@ SearchParameters.prototype = {
     var newState = new ( this.constructor )( this );
     fn( newState );
     return Object.freeze( newState );
+  },
+  /**
+   * Let the user set a specific value for a given parameter. Will return the
+   * same instance if the parameter is invalid or if the value is the same as the
+   * previous one.
+   * @method
+   * @param {string} parameter the parameter name
+   * @param {any} value the value to be set, must be compliant with the definition of the attribute on the object
+   * @return {SearchParameters} the updated state
+   */
+  setQueryParameter : function setParameter( parameter, value ) {
+    var k = keys( this );
+    if( k.indexOf( parameter ) === -1 ) return this;
```

Comment:
If I was using this I would really want it to throw if I am passing an unknown argument. If the goal is to only accept known arguments then it should throw. Logging would be ok if it was not an error.

But the client will expect something to work and it will not so it seems like it's not a warning.

You could also emit an error and if he does not listen to the error event of an EventEmitter it will throw (default behavior).

But then just throwing seems better to me.